### PR TITLE
JAMES-2080 Address headers do not require NESTED ElasticSearch mapping

### PR DESCRIPTION
**Pull request by chibenwa**
Nested objects are indexed as separate documents which incurs an overhead
at index time and also has a cost upon search as ElasticSearch needs to
perform joins between the nested documents and the original one.

It turns out that the extra coherence between mailbox name and address
brought by nested objects is not needed functionally speaking and thus we
can get rid of it easily.

Note that this change needs a re-indexing of previously existing data

---------------------------------------------------

Status of this work: we need to:

 - Evaluate it against performance criteria
 - Will be merged alongside other beneficial ES mapping enhancements

### CODE REVIEW

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java [hunk 1]
```diff
@@ -318,10 +319,20 @@ private IndexableMessage(
...
+        return bcc.getEmailers().stream()
+            .map(EMailer::getName)
+            .collect(Guavate.toImmutableSet());
+    }
+
+    @JsonProperty(JsonMessageConstants.BCC + JsonMessageConstants.EMailer.ADDRESS)
+    public Set<String> getBccAddress() {
+        return bcc.getEmailers().stream()
+            .map(EMailer::getAddress)
+            .collect(Guavate.toImmutableSet());
```
Comment (remk) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java]: You may extract a method `getEmailersProperty` taking into parameter and `EMailers` and the function in the map call.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`JamesMailSpoolerTest.threadSuicideShouldAckTheItem:126 » ConditionTimeout Cond...`

test this please

**Comment by chibenwa** [comment]
`ConcurrentTestRunnerTest.runShouldPerformAllOperationsEvenOnOccasionalExceptions:230 » NotTerminated`

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
I emmit doubt this changeset is worth it:

```
22.51 message indexed per second
6604 B per message (mean)
Mean 107 ms
P99 397 ms
```

As a record master branch was:

```
Reindexed/s:  22.46
Size per message: 5676 B
Mean: 108 ms
P99: 357 ms
```

As off today there is no clear evidence, given the tests conducted, that this is an improvment.

**Comment by chibenwa** [comment]
Not convincing

**Pull request closed without merging**